Refactor all map pages to be Laravel pages with AJAX data refresh

eskyuu opened the pull request:
This patch adds 2 new options, which are disabled by default, to allow all devices to be added to the map in the device overview page, and also to allow device dependencies to be shown on the same map.

It does touch the Maps DeviceDependencyController to add a couple of additional filters.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

eskyuu committed:
Added the option to show all devices on the device overview map

 - Created a new JSON query to fetch device data
 - Added configuration options to show devices and device links to the map
 - Updated the Map\DeviceDependency controller to allow for some more filters

eskyuu commented:
While looking at the way the various device maps have been implemented, it looks like all of the maps have been implemented differently.  If I did some work to standardise the way that all the maps get their data, would there be interest in merging the change?

eskyuu committed:
Formatting fixes

eskyuu committed:
Formatting fixes

eskyuu committed:
formatting fixes

eskyuu committed:
formatting fixes

eskyuu commented:
I believe that there is an error with the test script.  The YAML for the test that is failing works correctly as a boolean, and does not work if I convert it to a string.

murrant commented:
You'll need to add bool to the allowed types in the config schema here: misc/config_schema.json:54

Can you include screenshots? I don't understand what you are trying to accomplish.

Yes, unified code would be accepted.

eskyuu committed:
Allow boolean conditions in the settings page

eskyuu commented:
Thanks for the pointer regarding the bool option.

This commit affects the map when you click on the highlighted button on the device overview page below:
![image](https://github.com/librenms/librenms/assets/16013255/48ab3563-f4d4-4422-aaf0-d977f3998d87)

When both options are enabled, you will get something like the map below, showing the current device in the center, and links to dependent devices (or groups of devices if they have been grouped).
![image](https://github.com/librenms/librenms/assets/16013255/692c3e37-654b-4392-9e40-925e3a3b2171)

You can pan and zoom the same as the fullscreen map, which allows you to see if an issue is geographic.  Below is another example from our network after zooming out.  For someone unfamiliar with the network, they could zoom out and see if an issue at one location is related to an upstream issue.
![image](https://github.com/librenms/librenms/assets/16013255/47cc6ca0-ac93-470d-a2d0-5cc32c7d32ef)


If the blue lines don't look good they can be turned off, and both options are disabled by default.

eskyuu commented:
I'll have a look at the other map functions (both full screen and widgets) to get them to use a similar data load method, then I was planning on doing something similar with all the vis.js maps.

My end goal is to add a custom map option where people can build something similar to the Device Group maps, but with explicit device inclusion, map placement and link utilisation, as well as the ability to load a custom background (similar to the weathermap plugin).  I'm guessing that if I can unify the way the various maps load their data, it could make adding another map type easier to implement and more likely to be accepted.

eskyuu committed:
Full screen map update

 - Updated the full screen map to use the new layout, and also the new queries for the device and link lists.
 - The new code updated the items on the map without refreshing the entire page
 - I have also moved most of the new code into the MapController instead of DeviceDependencyController
 - English config page hints have been added for the new config options
 - The grouping radius for the device overview maps has been changed to use the configured option

eskyuu committed:
Full screen map update

 - Updated the full screen map to use the new layout, and also the new queries for the device and link lists.
 - The new code updated the items on the map without refreshing the entire page
 - I have also moved most of the new code into the MapController instead of DeviceDependencyController
 - English config page hints have been added for the new config options
 - The grouping radius for the device overview maps has been changed to use the configured option

eskyuu commented:
I spent yesterday afternoon re-working the full screen map to use the same device fetching code, and similar javascript code to fetch and update the devices.  As far as I could see, the map produces the same output as the original code, but I've made the refresh happen with an AJAX query (so if you have zoomed to a specific point on the map, it will just update the items on your screen instead of refreshing the whole page and returning you to the default location and zoom).

I've merged it into this request because it modified some of the same files.  Is this OK, or should I submit these as separate merge requests?

eskyuu committed:
Formatting fixes

eskyuu committed:
Formatting fixes

eskyuu committed:
More formatting fixes

eskyuu committed:
More formatting fixes

eskyuu committed:
Changed the full screen map refresh interval to use configure value

eskyuu committed:
Allow boolean when conditions

eskyuu committed:
Fix to really allow boolean when conditions

eskyuu committed:
Updated world map widget to use the device list query
Added blue icons to all map pages for devices under maintenance

eskyuu committed:
Migrated the availabiltiy map to the new page format using the background data load

eskyuu committed:
Migrated the availabiltiy map to the new page format using the background data load

PipoCanaja commented:
I would keep all in this PR, and change the description of the PR to something like "refactor and extend the device overview map". With details of the exact changes and screen captures in the description of the PR.

kkrumm1 commented:
nice work!

eskyuu committed:
Imrpoved the fullscreen check code on the world map page

eskyuu committed:
Update device dependency pages use Laravel with AJAX refresh

eskyuu committed:
Update device dependency pages use Laravel with AJAX refresh

eskyuu committed:
Remove unneeded PHP include

eskyuu committed:
Remove unneeded PHP include

eskyuu committed:
Re-added the original fullscreen check in addition to the new one for compatibility

eskyuu committed:
Converted all device map pages to Laravel

eskyuu committed:
Converted all device map pages to Laravel

eskyuu committed:
Added device formatting to new neighbour blade

eskyuu committed:
Formatting fixes

eskyuu committed:
Formatting fixes

eskyuu committed:
PHPStan fixes

eskyuu committed:
PHPStan fixes

eskyuu committed:
More PHPStan fixes

eskyuu committed:
More PHPStan fixes

eskyuu committed:
more PHPStan fixes

eskyuu committed:
Update metadata on new controller

eskyuu committed:
Update metadata on new controller

eskyuu commented:
As far as I'm concerned, this is ready to merge.  All pages should produce output that is the same as the original versions, so I'm not sure what use any more before and after shots would be.

There are improvements to all the dedicated map pages to make them refresh the items on the map without refreshing the whole page, so you will not lose the position that you have zoomed into.  There is also the original improvement of adding the option to show device links on the device overview map.

eskyuu committed:
Remove location marker when device links are shown on the map due to UI usability issues

eskyuu committed:
Added a link on the device overview map to the full screen map

eskyuu committed:
Formatting fixes

eskyuu committed:
Formatting fixes

eskyuu commented:
The last commit adds the icon in the top-right which takes you from the device overview page to the fullscreen map page:
![image](https://github.com/librenms/librenms/assets/16013255/37b53f1a-56f2-4a0d-a90c-ed55ae1ebc54)

eskyuu committed:
Hide the group list on availability map page depending on config

Jellyfrog commented on app/Http/Controllers/Device/Tabs/NeighboursController.php:
```diff
@@ -52,6 +55,86 @@ public function name(): string
+        $devices = [];
+
+        if ($selection == 'list') {
+            $linkQuery = Port::select(
```
You're fighting against the framework. 
Add a relationship for links here https://github.com/librenms/librenms/blob/master/app/Models/Port.php#L359
And query it like device instead

eskyuu replied:
I assumed that there was a reason that the links hadn't been added to the framework.  If this is incorrect, I will have a look at adding the links to the model.  This will also resolve both the items you've raised below which are workarounds for not having the relationships defined in the framework.

Can you please confirm that there isn't any reason you're aware of that would stop the links from being added as a model?  Is there something else that I need to clean up (e.g. an old database schema definition), or anything I need to be careful of to avoid a loss of existing data?  Is this a trickier than normal relationship (e.g. where the remote device id is allowed to be null), and possibly something I will need help with to get correct?

Jellyfrog commented on app/Http/Controllers/Device/Tabs/NeighboursController.php:
```diff
@@ -52,6 +55,86 @@ public function name(): string
+            }
+
+            foreach ($linkQuery->get() as $port) {
+                $row = json_decode(json_encode($port));
```
This is not necessary

Jellyfrog commented on app/Http/Controllers/Device/Tabs/NeighboursController.php:
```diff
@@ -52,6 +55,86 @@ public function name(): string
+
+                $rport = null;
+                if ($row->remote_port_id) {
+                    $rport = Port::where('port_id', '=', $row->remote_port_id)->with('device')->first();
```
This create n+1 queries. You should include it in big query; https://laravel.com/docs/10.x/eloquent-relationships#nested-eager-loading

eskyuu replied:
Please note that the existing page that this replaces (includes/html/pages/device/ports/neighbours.inc.php) calls device_by_id_cache() once for each neighbour, so this new page will have the same number of SQL queries as the original page.  The ideal solution would be to add links to the DB model, but I can't see why replacing a page with another that has the same number of queries would be an issue if the better solution is too tricky.

murrant replied:
device_by_id_cache() calls DeviceCache::get() in the background, which will only fetch each device once.
The best way would be to add a relationship to the Ports model to represent neighbor ports.

kkrumm1 commented:
I tried this PR - It breaks my world map. The device page still shows the map but no links

eskyuu committed:
Improve fullscreen check for world map

eskyuu committed:
Improve fullscreen check for world map

eskyuu commented:
> > > I tried this PR - It breaks my world map. The device page still shows the map but no links
> > 
> > 
> > I needed to run "php artisan route:cache" as the librenms user to get the new routes working for the background pages. Can you please confirm that you've done this. If the new routes are not added, the map will load, but the background requests for the devices and links will fail and the map will be blank.
> > Can you also confirm whether it's the full screen map that you're testing, or the map on the device overview page? When you say that the map still shows but no links, are you saying that the map works and the device shows on the map, but the links do not show, or that the map displays but no devices or links show on the map? If it is the second option, can you confirm that you enabled the device links in the settings page?
> 
> OK - I ran it again with php artisan route:cache - Now this is what happens
> 
> FullscreenMap shows only one node and no other nodes. But does show the links.
> 
> ![Screenshot 2023-11-30 at 4 00 18 PM](https://private-user-images.githubusercontent.com/27959419/287082038-829be99d-ce3a-4cf3-b850-2a1ddc258954.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzODc3NTUsIm5iZiI6MTcwMTM4NzQ1NSwicGF0aCI6Ii8yNzk1OTQxOS8yODcwODIwMzgtODI5YmU5OWQtY2UzYS00Y2YzLWI4NTAtMmExZGRjMjU4OTU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDIzMzczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkMzQyNzhmZDg1N2RiZTRmYWFhZTZjYjNkMjBiZTUzNzUzOGQ5Nzc1ZTdhNTRhNGI3OGY3ODc3NzYzYmJiNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZOYIwKOhEspoJqNc4bPgmpq_UXjPQhP6IFrAzwW4FMk)
> 
> The world map widget breaks in the dashboard. Its the widget on the top right.
> 
> ![Screenshot 2023-11-30 at 4 02 10 PM](https://private-user-images.githubusercontent.com/27959419/287082526-07f6d533-8b34-4da4-bbd6-db52042ffc43.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzODc3NTUsIm5iZiI6MTcwMTM4NzQ1NSwicGF0aCI6Ii8yNzk1OTQxOS8yODcwODI1MjYtMDdmNmQ1MzMtOGIzNC00ZGE0LWJiZDYtZGI1MjA0MmZmYzQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDIzMzczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkOTg2NzM0ZmRkODcxNDcwNDZlZjk4MzY4NzE1ZDkwNDA1YTRlOTEzYzZmZjA4YjM5MzE4YjAxODc5ZTA1MzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0xrlXf7QoHoMsaxLPzhO5aNma1Uymyu8AhjP6y3PON4)
> 
> The device page shows the map and device but none of the links show.
> 
> ![Screenshot 2023-11-30 at 4 03 01 PM](https://private-user-images.githubusercontent.com/27959419/287082833-02f7ad60-6245-4fd0-815a-725665d7845c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzODc3NTUsIm5iZiI6MTcwMTM4NzQ1NSwicGF0aCI6Ii8yNzk1OTQxOS8yODcwODI4MzMtMDJmN2FkNjAtNjI0NS00ZmQwLTgxNWEtNzI1NjY1ZDc4NDVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDIzMzczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3MWNmOTJlYjFmMDA3MzIzZmUwNzIzZGE2NjdiZTU5YmM4YTlhODgwYzlkYTY4NWQ1OWI4YzRjY2Y3YjljNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6YNwtZaJ9QKPbK1R4F4LTtdA721YYNiyuZM_4YhW0sY)

The fullscreen map output doesn't make sense to me.  If it's loading a single device, it should be loading all devices onto the map.  Are any errors logged to the librenms log file?

The world map widget is similar - it should at least show the map even if the devices fail to load.  Once again, are there any errors logged?

For the device overview page - have you enabled the appropriate options in the settings page?  These are the "show devices" and "show device dependencies" as shown below (I have also just noticed the typo in dependencies):
![image](https://github.com/librenms/librenms/assets/16013255/9ceb1587-772f-4489-8a0a-a464002eea15)

kkrumm1 commented:
> > > > I tried this PR - It breaks my world map. The device page still shows the map but no links
> > > 
> > > 
> > > I needed to run "php artisan route:cache" as the librenms user to get the new routes working for the background pages. Can you please confirm that you've done this. If the new routes are not added, the map will load, but the background requests for the devices and links will fail and the map will be blank.
> > > Can you also confirm whether it's the full screen map that you're testing, or the map on the device overview page? When you say that the map still shows but no links, are you saying that the map works and the device shows on the map, but the links do not show, or that the map displays but no devices or links show on the map? If it is the second option, can you confirm that you enabled the device links in the settings page?
> > 
> > 
> > OK - I ran it again with php artisan route:cache - Now this is what happens
> > FullscreenMap shows only one node and no other nodes. But does show the links.
> > The world map widget breaks in the dashboard. Its the widget on the top right.
> > The device page shows the map and device but none of the links show.
> 
> The fullscreen map output doesn't make sense to me. If it's loading a single device, it should be loading all devices onto the map. Are any errors logged to the librenms log file? Can you also please confirm what the network_map_items is set to on your librenms instance.
> 
> The world map widget is similar - it should at least show the map even if the devices fail to load. Once again, are there any errors logged? Can you also please confirm that the missing items show up on the pages when this PR is not applied. Some of the new functionality does not show up unless you enable it? Can you also please confirm what web browser you are using?
> 
> For the device overview page - have you enabled the appropriate options in the settings page? These are the "show devices" and "show device dependencies" as shown below (I have also just noticed the typo in dependencies): ![image](https://private-user-images.githubusercontent.com/16013255/287193112-9ceb1587-772f-4489-8a0a-a464002eea15.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0NTAwMzMsIm5iZiI6MTcwMTQ0OTczMywicGF0aCI6Ii8xNjAxMzI1NS8yODcxOTMxMTItOWNlYjE1ODctNzcyZi00NDg5LThhMGEtYTQ2NDAwMmVlYTE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDE2NTUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg5YzNiMjM1MzEwNzc1MWI2NTE3NDgxMWJkZWZiMGIwYmJmNjQ5ZTVmZmYwODg3YThlMGEwMmMxOWFiMDkyNzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oFgGJYjIBrQc1HqeXXxvzSTpZ-c8486GvvPjdA_gFVU)

FullScreenMap Settings are on - Tired with both CDP/LLDP and Dependencies.

![Screenshot 2023-12-01 at 10 57 13 AM](https://github.com/librenms/librenms/assets/27959419/69f324d6-f7bf-4249-9570-5e2a0f10b71c)
 
World Map widget is still blank -  Using Google Chrome
```
I see this in the log pop up
#13 /opt/librenms/vendor/symfony/console/Application.php(1081): Illuminate\Console\Command->run()
#14 /opt/librenms/vendor/symfony/console/Application.php(320): Symfony\Component\Console\Application->doRunCommand()
#15 /opt/librenms/vendor/symfony/console/Application.php(174): Symfony\Component\Console\Application->doRun()
#16 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#17 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#18 /opt/librenms/lnms(40): App\Console\Kernel->handle()
#19 {main}
```
The Device overview pages work now - I forgot about those settings 

![Screenshot 2023-12-01 at 10 57 13 AM](https://github.com/librenms/librenms/assets/27959419/a2d28ff3-8133-4154-a2f3-dbf9dba1bfc1)

murrant commented:
I made this if it helps: #15611

eskyuu commented:
> I made this if it helps: #15611

Thanks - I'll have a look at using that.

eskyuu commented:
> > > > > I tried this PR - It breaks my world map. The device page still shows the map but no links
> > > > 
> > > > 
> > > > I needed to run "php artisan route:cache" as the librenms user to get the new routes working for the background pages. Can you please confirm that you've done this. If the new routes are not added, the map will load, but the background requests for the devices and links will fail and the map will be blank.
> > > > Can you also confirm whether it's the full screen map that you're testing, or the map on the device overview page? When you say that the map still shows but no links, are you saying that the map works and the device shows on the map, but the links do not show, or that the map displays but no devices or links show on the map? If it is the second option, can you confirm that you enabled the device links in the settings page?
> > > 
> > > 
> > > OK - I ran it again with php artisan route:cache - Now this is what happens
> > > FullscreenMap shows only one node and no other nodes. But does show the links.
> > > The world map widget breaks in the dashboard. Its the widget on the top right.
> > > The device page shows the map and device but none of the links show.
> > 
> > 
> > The fullscreen map output doesn't make sense to me. If it's loading a single device, it should be loading all devices onto the map. Are any errors logged to the librenms log file? Can you also please confirm what the network_map_items is set to on your librenms instance.
> > The world map widget is similar - it should at least show the map even if the devices fail to load. Once again, are there any errors logged? Can you also please confirm that the missing items show up on the pages when this PR is not applied. Some of the new functionality does not show up unless you enable it? Can you also please confirm what web browser you are using?
> > For the device overview page - have you enabled the appropriate options in the settings page? These are the "show devices" and "show device dependencies" as shown below (I have also just noticed the typo in dependencies): ![image](https://private-user-> 
>
> FullScreenMap Settings are on - Tired with both CDP/LLDP and Dependencies.
> 
> World Map widget is still blank - Using Google Chrome
> 
> ```
> I see this in the log pop up
> #13 /opt/librenms/vendor/symfony/console/Application.php(1081): Illuminate\Console\Command->run()
> #14 /opt/librenms/vendor/symfony/console/Application.php(320): Symfony\Component\Console\Application->doRunCommand()
> #15 /opt/librenms/vendor/symfony/console/Application.php(174): Symfony\Component\Console\Application->doRun()
> #16 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
> #17 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
> #18 /opt/librenms/lnms(40): App\Console\Kernel->handle()
> #19 {main}
> ```
> 
> The Device overview pages work now - I forgot about those settings
> 
I had another thought on the worldmap - does the user viewing the map have full access to all devices, or is it a limited access user?  If it is a user with limited access, do they have permissions to view the missing devices?  I'm not sure about whether the original queries checked permissions, but the new queries will all only return hosts that the user has access to.

For the world map - if the error that you've shown happens every time the world map refreshes, please include all entries to #1.  I think those log entries are related to something run on the command line, and not the web.

eskyuu committed:
Alter full screen detection for world map

eskyuu committed:
Alter full screen detection for world map

kkrumm1 commented:
Here are the errors I see in the developer mode. 

keep seeing jquery errors on the map widget.
Also, see error on the weather map for http not https but i think that is separate.


```
overview:88 Flasher is not loaded
renderOptions @ overview:88
(anonymous) @ overview:88
VM6645:40 Uncaught SyntaxError: Unexpected token '}'
    at b (jquery.min.js?ver=05072021:2:866)
    at He (jquery.min.js?ver=05072021:2:48373)
    at S.fn.init.append (jquery.min.js?ver=05072021:2:49724)
    at S.fn.init.<anonymous> (jquery.min.js?ver=05072021:2:50816)
    at $ (jquery.min.js?ver=05072021:2:32425)
    at S.fn.init.html (jquery.min.js?ver=05072021:2:50494)
    at Object.success (overview:1650:34)
    at c (jquery.min.js?ver=05072021:2:28327)
    at Object.fireWith [as resolveWith] (jquery.min.js?ver=05072021:2:29072)
    at l (jquery.min.js?ver=05072021:2:79901)
b @ jquery.min.js?ver=05072021:2
He @ jquery.min.js?ver=05072021:2
append @ jquery.min.js?ver=05072021:2
(anonymous) @ jquery.min.js?ver=05072021:2
$ @ jquery.min.js?ver=05072021:2
html @ jquery.min.js?ver=05072021:2
success @ overview:1650
c @ jquery.min.js?ver=05072021:2
fireWith @ jquery.min.js?ver=05072021:2
l @ jquery.min.js?ver=05072021:2
(anonymous) @ jquery.min.js?ver=05072021:2
load (async)
send @ jquery.min.js?ver=05072021:2
ajax @ jquery.min.js?ver=05072021:2
widget_reload @ overview:1638
grab_data @ overview:1666
(anonymous) @ VM6608:1
b @ jquery.min.js?ver=05072021:2
He @ jquery.min.js?ver=05072021:2
append @ jquery.min.js?ver=05072021:2
S.fn.<computed> @ jquery.min.js?ver=05072021:2
h.add_widget @ jquery.gridster.min.js?ver=05072021:1
widget_dom @ overview:1566
(anonymous) @ overview:1283
each @ jquery.min.js?ver=05072021:2
(anonymous) @ overview:1282
overview:1 Mixed Content: The page at 'https://nms00.nbisd.org/overview' was loaded over HTTPS, but requested an insecure element 'http://nms00.nbisd.org/plugins/Weathermap/output/campuscore.png?size=1146.4666x1090.6666&924810045'. This request was automatically upgraded to HTTPS, For more information see https://blog.chromium.org/2019/10/no-more-mixed-messages-about-https.html
VM6649:40 Uncaught SyntaxError: Unexpected token '}'
    at b (jquery.min.js?ver=05072021:2:866)
    at He (jquery.min.js?ver=05072021:2:48373)
    at S.fn.init.append (jquery.min.js?ver=05072021:2:49724)
    at S.fn.init.<anonymous> (jquery.min.js?ver=05072021:2:50816)
    at $ (jquery.min.js?ver=05072021:2:32425)
    at S.fn.init.html (jquery.min.js?ver=05072021:2:50494)
    at Object.success (overview:1650:34)
    at c (jquery.min.js?ver=05072021:2:28327)
    at Object.fireWith [as resolveWith] (jquery.min.js?ver=05072021:2:29072)
    at l (jquery.min.js?ver=05072021:2:79901)
b @ jquery.min.js?ver=05072021:2
He @ jquery.min.js?ver=05072021:2
append @ jquery.min.js?ver=05072021:2
(anonymous) @ jquery.min.js?ver=05072021:2
$ @ jquery.min.js?ver=05072021:2
html @ jquery.min.js?ver=05072021:2
success @ overview:1650
c @ jquery.min.js?ver=05072021:2
fireWith @ jquery.min.js?ver=05072021:2
l @ jquery.min.js?ver=05072021:2
(anonymous) @ jquery.min.js?ver=05072021:2
load (async)
send @ jquery.min.js?ver=05072021:2
ajax @ jquery.min.js?ver=05072021:2
widget_reload @ overview:1638
grab_data @ overview:1666
(anonymous) @ overview:1670
setTimeout (async)
grab_data @ overview:1669
(anonymous) @ VM6608:1
b @ jquery.min.js?ver=05072021:2
He @ jquery.min.js?ver=05072021:2
append @ jquery.min.js?ver=05072021:2
S.fn.<computed> @ jquery.min.js?ver=05072021:2
h.add_widget @ jquery.gridster.min.js?ver=05072021:1
widget_dom @ overview:1566
(anonymous) @ overview:1283
each @ jquery.min.js?ver=05072021:2
(anonymous) @ overview:1282
overview:1 Mixed Content: The page at 'https://nms00.nbisd.org/overview' was loaded over HTTPS, but requested an insecure element 'http://nms00.nbisd.org/plugins/Weathermap/output/campuscore.png?size=1146.4666x1090.6666&2004293373'. This request was automatically upgraded to HTTPS, For more information see https://blog.chromium.org/2019/10/no-more-mixed-messages-about-https.html
VM6650:40 Uncaught SyntaxError: Unexpected token '}'
    at b (jquery.min.js?ver=05072021:2:866)
    at He (jquery.min.js?ver=05072021:2:48373)
    at S.fn.init.append (jquery.min.js?ver=05072021:2:49724)
    at S.fn.init.<anonymous> (jquery.min.js?ver=05072021:2:50816)
    at $ (jquery.min.js?ver=05072021:2:32425)
    at S.fn.init.html (jquery.min.js?ver=05072021:2:50494)
    at Object.success (overview:1650:34)
    at c (jquery.min.js?ver=05072021:2:28327)
    at Object.fireWith [as resolveWith] (jquery.min.js?ver=05072021:2:29072)
    at l (jquery.min.js?ver=05072021:2:79901)
b @ jquery.min.js?ver=05072021:2
He @ jquery.min.js?ver=05072021:2
append @ jquery.min.js?ver=05072021:2
(anonymous) @ jquery.min.js?ver=05072021:2
$ @ jquery.min.js?ver=05072021:2
html @ jquery.min.js?ver=05072021:2
success @ overview:1650
c @ jquery.min.js?ver=05072021:2
fireWith @ jquery.min.js?ver=05072021:2
l @ jquery.min.js?ver=05072021:2
(anonymous) @ jquery.min.js?ver=05072021:2
load (async)
send @ jquery.min.js?ver=05072021:2
ajax @ jquery.min.js?ver=05072021:2
widget_reload @ overview:1638
grab_data @ overview:1666
(anonymous) @ overview:1670
setTimeout (async)
grab_data @ overview:1669
(anonymous) @ overview:1670
setTimeout (async)
grab_data @ overview:1669
(anonymous) @ VM6608:1
b @ jquery.min.js?ver=05072021:2
He @ jquery.min.js?ver=05072021:2
append @ jquery.min.js?ver=05072021:2
S.fn.<computed> @ jquery.min.js?ver=05072021:2
h.add_widget @ jquery.gridster.min.js?ver=05072021:1
widget_dom @ overview:1566
(anonymous) @ overview:1283
each @ jquery.min.js?ver=05072021:2
(anonymous) @ overview:1282
overview:1 Mixed Content: The page at 'https://nms00.nbisd.org/overview' was loaded over HTTPS, but requested an insecure element 'http://nms00.nbisd.org/plugins/Weathermap/output/campuscore.png?size=1146.4666x1090.6666&257499335'. This request was automatically upgraded to HTTPS, For more information see https://blog.chromium.org/2019/10/no-more-mixed-messages-about-https.html
VM6665:40 Uncaught SyntaxError: Unexpected token '}'
    at b (jquery.min.js?ver=05072021:2:866)
    at He (jquery.min.js?ver=05072021:2:48373)
    at S.fn.init.append (jquery.min.js?ver=05072021:2:49724)
    at S.fn.init.<anonymous> (jquery.min.js?ver=05072021:2:50816)
    at $ (jquery.min.js?ver=05072021:2:32425)
    at S.fn.init.html (jquery.min.js?ver=05072021:2:50494)
    at Object.success (overview:1650:34)
    at c (jquery.min.js?ver=05072021:2:28327)
    at Object.fireWith [as resolveWith] (jquery.min.js?ver=05072021:2:29072)
    at l (jquery.min.js?ver=05072021:2:79901)
b @ jquery.min.js?ver=05072021:2
He @ jquery.min.js?ver=05072021:2
append @ jquery.min.js?ver=05072021:2
(anonymous) @ jquery.min.js?ver=05072021:2
$ @ jquery.min.js?ver=05072021:2
html @ jquery.min.js?ver=05072021:2
success @ overview:1650
c @ jquery.min.js?ver=05072021:2
fireWith @ jquery.min.js?ver=05072021:2
l @ jquery.min.js?ver=05072021:2
(anonymous) @ jquery.min.js?ver=05072021:2
load (async)
send @ jquery.min.js?ver=05072021:2
ajax @ jquery.min.js?ver=05072021:2
widget_reload @ overview:1638
grab_data @ overview:1666
(anonymous) @ overview:1670
setTimeout (async)
grab_data @ overview:1669
(anonymous) @ overview:1670
setTimeout (async)
grab_data @ overview:1669
(anonymous) @ overview:1670
setTimeout (async)
grab_data @ overview:1669
(anonymous) @ VM6608:1
b @ jquery.min.js?ver=05072021:2
He @ jquery.min.js?ver=05072021:2
append @ jquery.min.js?ver=05072021:2
S.fn.<computed> @ jquery.min.js?ver=05072021:2
h.add_widget @ jquery.gridster.min.js?ver=05072021:1
widget_dom @ overview:1566
(anonymous) @ overview:1283
each @ jquery.min.js?ver=05072021:2
(anonymous) @ overview:1282
overview:1 Mixed Content: The page at 'https://nms00.nbisd.org/overview' was loaded over HTTPS, but requested an insecure element 'http://nms00.nbisd.org/plugins/Weathermap/output/campuscore.png?size=1146.4666x1090.6666&129788112'. This request was automatically upgraded to HTTPS, For more information see https://blog.chromium.org/2019/10/no-more-mixed-messages-about-https.html
```

kkrumm1 commented:
Here is a better output with only the map widget on the dashboard. 
```

Fetch event handler is recognized as no-op. No-op fetch handler may bring overhead during navigation. Consider removing the handler if possible.
6:88 Flasher is not loaded
renderOptions @ 6:88
6:40 Uncaught SyntaxError: Unexpected token '}'
    at b (jquery.min.js?ver=05072021:2:866)
    at He (jquery.min.js?ver=05072021:2:48373)
    at S.fn.init.append (jquery.min.js?ver=05072021:2:49724)
    at S.fn.init.<anonymous> (jquery.min.js?ver=05072021:2:50816)
    at $ (jquery.min.js?ver=05072021:2:32425)
    at S.fn.init.html (jquery.min.js?ver=05072021:2:50494)
    at Object.success (6:1650:34)
    at c (jquery.min.js?ver=05072021:2:28327)
    at Object.fireWith [as resolveWith] (jquery.min.js?ver=05072021:2:29072)
    at l (jquery.min.js?ver=05072021:2:79901)
```

eskyuu committed:
Fixed up map view when showing devices with disabled alerts

eskyuu committed:
Fixed up map view when showing devices with disabled alerts

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/librenms/librenms?pullRequest=15567) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/librenms/librenms?pullRequest=15567) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: eskyuu<br/>:x: NMS User<br/><hr/>**NMS User** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/librenms/librenms?pullRequest=15567) it.</sub>

eskyuu committed:
Formatting fixes

eskyuu committed:
Formatting fixes

eskyuu commented:
> Here is a better output with only the map widget on the dashboard.

Sorry for the delay in replying.  I have re-configured our systems to allow me to run custom branches properly, and I was able to replicate the issues you were facing.  I found that having no device group on the widget, and showing disabled devices on the full screen map replicated the issues.  I have applied fixes for these 2 conditions.

Can you please try the updated code and let me know if it's working correctly for you now?

kkrumm1 commented:
That fixed it! Everything is working now. I think this PR should be good to go. See if anybody has objections.

eskyuu committed:
Update device neighbour tab and controller to use Link model

eskyuu committed:
Remove reference to unused model

eskyuu committed:
Remove reference to unused model

eskyuu committed:
Split map controller into separate files and update routes

eskyuu committed:
Convert all data requests to POST queries
Fixed a route typo
Added missing dependency to device dependecy controller

eskyuu committed:
Fixed a couple of unneeded dependencies

eskyuu committed:
Removed Map Controller

eskyuu committed:
Removed Map Controller

eskyuu committed:
Fixed NetMap Controller dependency and vis option config

eskyuu committed:
Convert geo link query to Laravel

eskyuu committed:
Formattinx fixes

eskyuu committed:
Formatting fixes

eskyuu committed:
Added a view and got the MAC address links working through the view and with correct Laravel queries

eskyuu committed:
Added a view and got the MAC address links working through the view and with correct Laravel queries

eskyuu committed:
Query fixes when getting MAC links

eskyuu committed:
Added filters to the list of device loaded
Made the remote port lst attribute dynamic in preparation for adding XDP links

eskyuu committed:
Formatting fixes

eskyuu committed:
Fix validation error with create view

eskyuu committed:
Fix validation error with create view

eskyuu committed:
Re-added XDP links using the links table

eskyuu committed:
Re-added XDP links using the links table

eskyuu committed:
Improve the ARP table lookup using the new MAC address views

eskyuu committed:
Formatting fixes

eskyuu committed:
Remove trailing whitespace

eskyuu committed:
DB Schema Update

eskyuu committed:
Update view creation migration

eskyuu committed:
Update view creation migration

eskyuu committed:
Another attempt at the view migration to be more compatible with all mysql and mariadb versions

eskyuu committed:
Another attempt at the view migration to be more compatible with all mysql and mariadb versions

eskyuu committed:
change back to CREATE OR REPLACE VIEW

eskyuu committed:
change back to CREATE OR REPLACE VIEW

eskyuu committed:
Update view migration to drop then create view to make it compatible with all DB version

eskyuu committed:
Update view migration to drop then create view to make it compatible with all DB version

eskyuu commented:
I knew that I had a couple of places in this code where I was not using relationships correctly.  I have fixed these today, and also created a view (and associated model) that allows us to find remote ports that match the IP and MAC address of an ARP entry of the local port.

Can I please get comments on suitability for accepting this request as-is.

I know that I have modified an old style page to use new style queries.  I think that given it's part of the original ajax table queries, it is a step forward even though the rest of the queries are not ready for migration.

eskyuu committed:
Update device overview map to use POST to fetch data

eskyuu committed:
Formatting fixes

eskyuu committed:
Formatting fixes

eskyuu committed:
Remove full screen test on world map

eskyuu committed:
Remove full screen test on world map

eskyuu committed:
Added namespace back into maps route

eskyuu commented:
Any takers to review this and provide feedback?

electrocret commented:
@eskyuu  - this needs more work. 

Fullscreen Geographical Map returns a not found.
Device Dependencies has an error.


tail logs/librenms.log -n 40
<span class=sf-dump-str title=\"1357 characters\">      &quot;damping&quot;: 0.4,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;avoidOverlap&quot;: 1<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">    },<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\"><span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">     &quot;repulsion&quot;: {<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;centralGravity&quot;: 0.2,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;springLength&quot;: 250,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;springConstant&quot;: 0.2,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;nodeDistance&quot;: 200,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;damping&quot;: 0.07<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">    },<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\"><span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">     &quot;hierarchicalRepulsion&quot;: {<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;nodeDistance&quot;: 300,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;centralGravity&quot;: 0.2,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;springLength&quot;: 300,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;springConstant&quot;: 0.2,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">      &quot;damping&quot;: 0.07<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">    },<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\"><span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">  &quot;maxVelocity&quot;: 50,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">  &quot;minVelocity&quot;: 0.4,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">  &quot;solver&quot;: &quot;hierarchicalRepulsion&quot;,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">  &quot;stabilization&quot;: {<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">    &quot;enabled&quot;: true,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">    &quot;iterations&quot;: 1000,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">    &quot;updateInterval&quot;: 100,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">    &quot;onlyDynamicEdges&quot;: false,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">    &quot;fit&quot;: true<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">  },<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\"><span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">  &quot;timestep&quot;: 0.4,<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\"> }<span class=\"sf-dump-default sf-dump-ns\">\\n</span></span>
<span class=sf-dump-str title=\"1357 characters\">}</span>
\"\"\"
</pre><script>Sfdump(\"sf-dump-1664753658\", {\"maxDepth\":3,\"maxStringLength\":160})</script>
","group_name":"<pre class=sf-dump id=sf-dump-2130632329 data-indent-pad=\"  \"><span class=sf-dump-const>null</span>
</pre><script>Sfdump(\"sf-dump-2130632329\", {\"maxDepth\":3,\"maxStringLength\":160})</script>
"}},"userId":7,"exception":"[object] (Spatie\\LaravelIgnition\\Exceptions\\ViewException(code: 0): Route [maps.getdevices] not defined. at /opt/librenms/vendor/laravel/framework/src/Illuminate/Routing/UrlGenerator.php:477)
[previous exception] [object] (Symfony\\Component\\Routing\\Exception\\RouteNotFoundException(code: 0): Route [maps.getdevices] not defined. at /opt/librenms/vendor/laravel/framework/src/Illuminate/Routing/UrlGenerator.php:477)"}

eskyuu commented:
@electrocret  - You need to run php artisan route:cache after checking out this branch / applying the patch.

electrocret commented:
> @electrocret  - You need to run "php artisan route:cache" after checking out this branch / applying the patch.
> 

Ah, I didn't have much time to dig into this today. I'll try again tomorrow.

electrocret commented:
Tested it again after following you direction. All the maps I tested loaded fine. However I noticed the auto-refresh is on for the fullscreen GeoMap, and appears to display.

![image](https://github.com/librenms/librenms/assets/6815956/028cace6-e145-4168-95d7-fdca732da8df)

eskyuu commented:
> Tested it again after following you direction. All the maps I tested loaded fine. However I noticed the auto-refresh is on for the fullscreen GeoMap, and appears to display.
> 
> Edit - just reverted back, and it looks like auto-refresh is on currently, however the count down is hidden in the menu.
> 
> ![image](https://private-user-images.githubusercontent.com/6815956/312214572-028cace6-e145-4168-95d7-fdca732da8df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyOTI5MzYsIm5iZiI6MTcxMDI5MjYzNiwicGF0aCI6Ii82ODE1OTU2LzMxMjIxNDU3Mi0wMjhjYWNlNi1lMTQ1LTQxNjgtOTVkNy1mZGNhNzMyZGE4ZGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTNUMDExNzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTYyN2U4MWQ2ZjVlYTc2Y2UyMDNjMzc4ZTMxMjk0MDJhOGJiMWJlZjdlYTRiZWYzMTE5MWNkMTlhYjAxMTJhMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.FN0E6lQ3m0VhpimoQAUATwO_Pk7IWeP6aQ6Dj0MkGlg)

Yes, I haven't been consistent with showing the refresh interval, but I believe that I made it visible for pages where there was an option to pause it in the menu.  This was done to maintain the same functionality with the new pages.  I'd be happy to show the refresh countdown with the pause button on all pages if it is considered helpful.

eskyuu committed:
Added the option to show all devices on the device overview map

 - Created a new JSON query to fetch device data
 - Added configuration options to show devices and device links to the map
 - Updated the Map\DeviceDependency controller to allow for some more filters

eskyuu committed:
Updated world map widget to use the device list query
Added blue icons to all map pages for devices under maintenance

eskyuu committed:
Remove location marker when device links are shown on the map due to UI usability issues

eskyuu committed:
Added a link on the device overview map to the full screen map

eskyuu committed:
Convert all data requests to POST queries
Fixed a route typo
Added missing dependency to device dependecy controller

eskyuu committed:
Change map behaviour so single click highlights and double-click opens the device

eskyuu committed:
Added option to highlight dependent devices in the device dependecy map

murrant commented:
@eskyuu dang, I dropped this PR.  Did you see I refactored the Map widget to use an ajax refresh recently?

eskyuu commented:
@murrant I did see that, so I dropped it from this PR.  I can add it back in if you think having a single endpoint for map data requests is a good idea.

eskyuu committed:
Converted the parent/child code to use Collecitons and added some types

eskyuu committed:
Fixed some types

eskyuu committed:
Convert all parent code to use Collections

eskyuu committed:
Change child highlight to work the same as parent

eskyuu committed:
Remove debug line

eskyuu committed:
bug fixes

eskyuu committed:
Formatting fixes

murrant commented:
If it isn't too bad a single endpoint would be better.  Can you adapt the code I added to support this case (or vice-versa)

eskyuu commented:
I'll add it back in and test

eskyuu committed:
Fixed some types

eskyuu committed:
Show devices with alerts disabled on various maps

eskyuu committed:
Use node disabled style for offline devices with alerts disabled

eskyuu committed:
Fix device dependency node highlight to select None instead of Isolated devices

eskyuu committed:
Fixed up top level device detection in the event of 2 devices only having each other as peers

eskyuu committed:
Added remaining types to functions

eskyuu committed:
Added includes for some types

eskyuu committed:
More type fixes

eskyuu committed:
Type fixes

eskyuu committed:
Remove unneeded use

eskyuu committed:
Update the world map widget to use the new map data controller

eskyuu committed:
fixed bug with map widget data fetch

murrant commented on app/Http/Controllers/Maps/MapDataController.php:
```diff
@@ -0,0 +1,799 @@
+
+        $group_id = $request->group;
+        if ($group_id) {
+            $linkQuery->join('device_group_device AS ldg', 'ld.device_id', '=', 'ldg.device_id')
```
Is it possible to use relationships for this query?  That would avoid the phpstan errors.

eskyuu replied:
How does this change look?

eskyuu committed:
Fix device group selection in full screen map

murrant reviewed (dismissed):
Tested ok, but is large enough it will break something :)

eskyuu commented:
Thanks!  We've been running this for about a yer now on our production systems, but I agree that there will very likely be some combination that I've not correctly accounted for.  Please highlight any bug reports to me - I'm very happy to help clean things up as needed.

murrant commented on app/Http/Controllers/Maps/FullscreenMapController.php:
```diff
@@ -0,0 +1,84 @@
+{
+    protected static function fullscreenMap(Request $request): View
+    {
+        $group_name = null;
```
Probably should add validation here

eskyuu replied:
done

murrant commented on database/migrations/2023_12_21_085427_create_view_port_mac_link.php:
```diff
@@ -0,0 +1,47 @@
+     */
+    public function up(): void
+    {
+        DB::statement('DROP VIEW IF EXISTS view_port_mac_links;');
```
Not sure how I feel about the addition of views.  I know not all databases support them, so if it is a goal to eventually support other databases, they will need to be removed then.

eskyuu replied:
Most databases that I'm aware of support simple views like this.  It does pass the MySQL and SQLite tests (I had to make changes to get it to work on the latter).

From memory this was needed to get a significant speed increase in doing the MAC+port lookup.  It has been a long time since I did this, so I'd need to re-discover the original issue to see if I can find another workaround.

murrant commented on html/js/librenms.js:
```diff
@@ -441,10 +417,10 @@ function destroy_map(id) {
-        type: "GET",
-        url: ajax_url + '/dash/worldmap',
+        type: "POST",
+        url: '/maps/getdevices',
```
Urls like this won't work when people use subdirectories.

eskyuu replied:
Fixed

murrant commented on includes/html/table/arp-search.inc.php:
```diff
@@ -1,121 +1,104 @@
+    ->withAggregate('device', 'hostname')
+    ->orderBy(...$sort_arr);
+
+if (! $user->hasGlobalRead()) {
```
Might be better to use the hasAccess scope

eskyuu replied:
Done

murrant commented on includes/html/table/arp-search.inc.php:
```diff
@@ -1,121 +1,104 @@
 <?php
```
If you are going to substantially re-write this it would be better to put it under a Laravel route (instead of a legacy one) Like App\Http\Controllers\Table

murrant commented on resources/views/device/tabs/neighbours.blade.php:
```diff
@@ -0,0 +1,101 @@
+var network = new vis.Network(container, data, options);
+network.on('click', function (properties) {
+    if (properties.nodes > 0) {
+       window.location.href = "device/device="+properties.nodes+"/tab=neighbours/selection=map/"
```
Suspect url here.

eskyuu replied:
Original code was in includes/html/print-map.inc.php and used this format.  I'm pretty sure it uses "base href=" in resources/views/layouts/librenmsv1.blade.php.

I've converted it to use @url()

murrant commented on routes/web.php:
```diff
@@ -84,13 +84,21 @@
+    Route::get('fullscreenmap', 'Maps\FullscreenMapController@fullscreenMap');
+    Route::get('availability-map', 'Maps\AvailabilityMapController@availabilityMap');
+    Route::get('map/{vars?}', 'Maps\NetMapController@netMap');
+    Route::prefix('maps')->namespace('Maps')->group(function () {
```
Instead of namespace Maps here, you could just use the newer array syntax.  I don't feel that strongly about it though.

eskyuu committed:
Allow maps url to work with sub-directories

eskyuu committed:
Converted URL reference to use @url

eskyuu committed:
Update arp-search to use hasAccess()

eskyuu committed:
Fixed fullscreen map init and removed unused input parameters

eskyuu commented:
@murrant - Sorry for the late changes here.  I didn't notice your requested changes after it was initially approved (before the last conflicting merge).  Other than the view, I thinks I've addressed all your major concerns.  Please get my attention if there are any other show-stoppers for this.

murrant commented on misc/db_schema.yaml:
```diff
@@ -2273,6 +2273,11 @@ vendor_ouis:
+  Columns:
+    - { Field: port_id, Type: 'int unsigned', 'Null': false, Extra: '', Default: '0' }
+    - { Field: ipv4_mac_id, Type: 'bigint unsigned', 'Null': false, Extra: '', Default: '0' }
+    - { Field: remote_port_id, Type: 'int unsigned', 'Null': false, Extra: '', Default: '0' }
```
Hmm, I don't think the db schema stuff can handle views.

murrant commented on html/js/librenms.js:
```diff
@@ -441,10 +417,10 @@ function destroy_map(id) {
 
 function populate_map_markers(map_id, group_radius = 10, status = [0,1], device_group = 0) {
```
Lets move this to worldmap.blade.php (I think that is the only usage), then we can use the route() helper to get the ajax url.

eskyuu replied:
done

eskyuu committed:
Move populate_map_markers() into fullscreen.blade.php

murrant committed:
use route helper

murrant committed:
Use route helper

murrant committed:
Move populate_map_markers to the correct location

murrant committed:
Exclude views from db schema dump for now
